ostd/src/task/scheduler/mod.rs

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#![allow(dead_code)]
+
+mod fifo_scheduler;
+
+pub use fifo_scheduler::FifoScheduler;
+use spin::Once;
+
+use super::{processor::switch_to_task, task::Task, TaskStatus};
+use crate::{arch::timer::register_callback, prelude::*};
+
+static SCHEDULER: Once<&'static dyn Scheduler<Task>> = Once::new();
+
+/// Injects a scheduler implementation into framework.
+///
+/// This function can only be called once and must be called during the initialization of kernel.
+pub fn inject_scheduler(scheduler: &'static dyn Scheduler) {
+    SCHEDULER.call_once(|| scheduler);
+
+    register_callback(|| {
```

Comment:
Again, do not import static functions directly, such as `arch::timer::register_callback`. Instead, import `arch::timer::self` and call `timer::register_callback` for improved clarity.

Reply:
I'll fix this as advised.